app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -688,8 +718,24 @@ public void checkIv() {
             }
         }
 
-        int selectedPokemon = pokemonList.getSelectedItemPosition();
-        Pokemon pokemon = pokeCalculator.get(selectedPokemon);
+
+        //below picks a pokemon from either the pokemon spinner or the user text input
+        Pokemon pokemon;
+        if (usingSpinnerPickerForPokemon) { //user picked pokemon from spinner
+            int selectedPokemon = pokemonList.getSelectedItemPosition();
+            pokemon = pokeCalculator.get(selectedPokemon);
+        } else { //user typed manually
+            String userInput = autoCompleteTextView1.getText().toString();
+            pokemon = pokeCalculator.get(userInput);
+            if (pokemon == null) { //no such pokemon was found, show error toast and abort showing results
+                Toast.makeText(this, userInput + getString(R.string.wrongPokemonNameInput), Toast.LENGTH_SHORT).show();
```

Comment:
This sort of thing is bad for translator—translation requires using format strings since the pokemon name might not be the first word in all languages.